api/src/test/java/org/openmrs/validator/ObsValidatorTest.java

```diff
@@ -513,4 +517,23 @@ public void validate_shouldPassValidationIfValueTextIsLessThanTheMaximumLength()
 		Assert.assertFalse(errors.hasFieldErrors("obsDatetime"));
 		Assert.assertFalse(errors.hasFieldErrors("valueText"));
 	}
+
+	@Test
+	public void validate_shouldFailValidationForDifferentObsPersonAndWithEncounterPatient() {
+		EncounterService es = Context.getEncounterService();
+		PersonService ps = Context.getPersonService();
+
+		Obs obs = new Obs(1);
+		obs.setPerson(ps.getPerson(2)); // Setting this person (2) as different from that of encounter person
```

Comment:
please remove all the comments as its clear from the test method name and the code that they are different